flyteadmin/pkg/repositories/config/seed_data.go

```diff
@@ -27,3 +30,349 @@ func SeedProjects(db *gorm.DB, projects []string) error {
 	}
 	return tx.Commit().Error
 }
+
+func ValidateMatcheableResourceLevel(project string, domain string, workflow string) error {
+	if project == "" {
+		return fmt.Errorf("project is required")
+	}
+	if domain == "" && workflow != "" {
+		return fmt.Errorf("domain is required: %s %s", project, workflow)
+	}
+	return nil
+}
+
+func CreateAttributeModel(project string, domain string, workflow string, attribute *admin.MatchingAttributes, resourceType admin.MatchableResource) (models.Resource, error) {
+	err := ValidateMatcheableResourceLevel(project, domain, workflow)
+	if err != nil {
+		return models.Resource{}, err
+	}
+
+	var model models.Resource
+
+	if workflow != "" {
+		model, err = transformers.WorkflowAttributesToResourceModel(
+			admin.WorkflowAttributes{
+				Project:            project,
+				Domain:             domain,
+				Workflow:           workflow,
+				MatchingAttributes: attribute,
+			},
+			resourceType,
+		)
+
+	} else if domain != "" {
+		model, err = transformers.ProjectDomainAttributesToResourceModel(
+			admin.ProjectDomainAttributes{
+				Project:            project,
+				Domain:             domain,
+				MatchingAttributes: attribute,
+			},
+			resourceType,
+		)
+	} else {
+		model, err = transformers.ProjectAttributesToResourceModel(
+			admin.ProjectAttributes{
+				Project:            project,
+				MatchingAttributes: attribute,
+			},
+			resourceType,
+		)
+
+	}
+	return model, err
+}
+
+func CreateKey(project string, domain string, workflow string, resourceType string) string {
+	return fmt.Sprintf("%s-%s-%s-%s", project, domain, workflow, resourceType)
```

Comment:
nit: should we version this string perhaps? (if we want to one day modify the format to support task or launch plan level overrides, for example) e.g. `v1-%s-%s...`